docs/built-in-functions.rst

```diff
@@ -0,0 +1,237 @@
+.. index:: function, built-in;
+
+.. _built_in_functions:
+
+***********************
+Built in Functions
+***********************
+
+Viper contains a set amount of built in functions that would be timely and/or unachievable to write in Viper.
+
+.. _functions:
+
+Functions
+=========
+The first list dictates each argument and the second sublist lists the types allowed as input.
+
+* **floor**
+::
+
+  def floor(a) -> b:
+    """
+    :param a: value to round down
+    :type a: either decimal or num
+
+    :output b: interger
+    """
+Rounds a decimal down to the nearest integer.
+
+* **decimal**
+::
+
+  def decimal(a) -> b:
+    """
+    :param a: value to turn into decimal
+    :type a: either decimal or num
+
+    :output b: decimal
+    """
+Turns a number into a decimal.
+
+* **as_unitless_number**
+::
+
+  def as_unitless_number(a) -> b:
+    """
+    :param a: value to remove units from
+    :type a: either decimal or num
+
+    :output b: either decimal or num
+    """
+Turns a ``num`` or ``decimal`` with units into one without units (used for assignment and math).
+
+* **as_num128**
+::
+
+  def as_num128(a) -> b:
+    """
+    :param a: value to turn into int128
+    :type a: either num, bytes32, num256, or bytes
```

Comment:
Consider specific examples as to how this transformation occurs, in order to avoid surprises and sneaky bugs for developers.